tracer/test/Datadog.Trace.Tests/DataStreamsMonitoring/DataStreamsContextPropagatorTests.cs

```diff
@@ -0,0 +1,34 @@
+﻿// <copyright file="DataStreamsContextPropagatorTests.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+
+using System;
+using Datadog.Trace.DataStreamsMonitoring;
+using Datadog.Trace.DataStreamsMonitoring.Hashes;
+using FluentAssertions;
+using Xunit;
+
+namespace Datadog.Trace.Tests.DataStreamsMonitoring;
+
+public class DataStreamsContextPropagatorTests
+{
+    [Fact]
+    public void CanRoundTripPathwayContext()
+    {
+        var headers = new TestHeadersCollection();
+        var context = new PathwayContext(
+            new PathwayHash(1234),
+            DateTimeOffset.UtcNow.ToUnixTimeMilliseconds() * 1000, // leaky abstraction, the encoder truncates
```

Comment:
Instead of incorporating the encoder truncation here, you could change your assertion at the bottom to have the assertion accept a value that is up to 1ms off?

Reply:
Done in https://github.com/DataDog/dd-trace-dotnet/pull/3174/commits/4552c91bccfe7019b7f687c1ffe9222f22712fba, I'm not _entirely_ sure it's easier to understand (as still need to do the conversion from nano to milli etc) but maybe it is! 😄